crates/starknet-os/src/execution/deprecated_syscall_handler.rs

```diff
@@ -273,17 +263,12 @@ where
                 HintError::SyscallError("No more storage reads available to replay".to_string().into_boxed_str()),
             )?;
 
-        vm.insert_value((syscall_ptr + 2usize).unwrap(), value).unwrap();
+        vm.insert_value((syscall_ptr + 2usize)?, value)?;
 
         Ok(())
     }
     pub async fn storage_write(&self, _syscall_ptr: Relocatable) -> Result<(), HintError> {
-        let sys_hand = self.deprecated_syscall_handler.write().await;
-
-        let _ = sys_hand.exec_wrapper.execution_helper.write().await.execute_code_read_iter.next().ok_or(
-            HintError::SyscallError("No more storage writes available to replay".to_string().into_boxed_str()),
-        )?;
-
+        // Nothing to do
```

Comment:
Good remark! Newer versions of Blockifier now split reads and writes into two separate fields, `CallInfo.storage_read_values` and `CallInfo.accessed_storage_keys`. Before the two were merged so we had to take a value from the iterator (writes were always set to 0, the value itself has no interest) so that the read syscall could take the correct values.

Now, we can just iterate over `storage_read_values` in the read syscall and do nothing in the write syscall. Technically we could iterate over the `accessed_storage_keys` in the write syscall but there is no interest in doing that besides checking that we have an equal amount of values. It sounds a bit useless to me so I just replaced this code by a no-op.